ui/src/plugins/dev.perfetto.RecordTraceV2/local_flag.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2025 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Flag} from '../../public/feature_flag';
+
+// Flag that indicates if a non-default socket should be used to connect to traced.
+// It is present in this file circular dependencies.
+export const LOCAL_ABSTRACT_SOCKET_FLAG: {flag?: Flag} = {};
```

Comment:
This is actually quite pragmatic. I'd just move also the registration here for consistency

```
const settings = {
  consumerSocket: ...,
}

function registerRecordingSettings(app: App) {
 app.settings.register(...)
}

getConsumerSockerFlag() {
   // Calling this before registerRecordingSettings() should never
   // happen and should just trigger a crash
   assertExists(settings.consumerSocket). ....
}
```

Reply:
I created a `settings` object which can be use to include all the settings of the `RecordTraceV2` plugin. 
It includes the registration as requested.